File: VB/RichEditDemo.Wpf/DemoUtils.vb
Imports System.IO
Imports DevExpress.DemoData.Helpers

Namespace RichEditDemo
    Public Class DemoUtils
        Public Shared Function GetRelativePath(ByVal name As String) As String
            name = "Data\" & name
            Dim path As String = DataFilesHelper.DataDirectory
            Dim s As String = "\"
            For i As Integer = 0 To 10
                If File.Exists(path & s & name) Then
                    Return System.IO.Path.GetFullPath(path & s & name)
                Else
                    s &= "..\"
                End If
            Next i
            Return ""
        End Function
    End Class
End Namespace

File: CS/RichEditDemo.Wpf/DemoUtils.cs
using System.IO;
using DevExpress.DemoData.Helpers;

namespace RichEditDemo {
    public class DemoUtils {
        public static string GetRelativePath(string name) {
            name = "Data\\" + name;
            string path = DataFilesHelper.DataDirectory;
            string s = "\\";
            for (int i = 0; i <= 10; i++) {
                if (File.Exists(path + s + name))
                    return Path.GetFullPath(path + s + name);
                else
                    s += "..\\";
            }
            return "";
        }
    }
}
